core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -164,15 +178,18 @@ public void readFrom(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeOptionalStreamable(internalShardId);
+        out.writeOptionalStreamable(shardId);
```

Comment:
I fixed it in the feature branch pulling in the changes to readFrom/writeTo methods already.  https://github.com/areek/elasticsearch/commit/d6df74baebd52d26e0200d818e0e65fec442a749 fixes this for this PR.